src/realm/object-store/sync/app.hpp

```diff
@@ -62,10 +64,8 @@ class App : public std::enable_shared_from_this<App>, public AuthRequestClient,
 
     // `enable_shared_from_this` is unsafe with public constructors; use `get_shared_app` instead
     App(const Config& config);
-    App(const App&) = default;
     App(App&&) noexcept = default;
-    App& operator=(App const&) = default;
```

Comment:
Why are we disallowing copy construction entirely as opposed to just implementing the actors with not copying over the mutex?

Reply:
Copying App seems semantically incorrect (there's only supposed to be one App per app id) and nothing was actually relying on being able to copy it.

Reply:
Makes sense.